sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLDriver.scala

```diff
@@ -50,8 +52,21 @@ private[hive] class SparkSQLDriver(val context: SQLContext = SparkSQLEnv.sqlCont
     if (analyzed.output.isEmpty) {
       new Schema(Arrays.asList(new FieldSchema("Response code", "string", "")), null)
     } else {
-      val fieldSchemas = analyzed.output.map { attr =>
-        new FieldSchema(attr.name, attr.dataType.catalogString, "")
+      val fieldSchemas = stripRootCommandResult(query.executedPlan) match {
+        case ExecutedCommandExec(s: ShowTablesCommand) if !s.isExtended =>
```

Comment:
are these code copied from somewhere?

Reply:
Yea, This(schema output) is going to be the same thing as [values output](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/HiveResult.scala#L64)

Reply:
instead of handling values and schema separately, shall we implement this handling as a catalyst rule so that we only need to do it once?

Reply:
This is a spark-sql cli printing problem. can we use catalyst rule to handle it uniformly?